src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -252,6 +477,226 @@ private TemporaryPoint createNewPoint(final Coordinate coordinate,
+    {
+        final Optional<IsoCountry> countryOne = ISOCountryTag.first(one);
+        final Optional<IsoCountry> countryTwo = ISOCountryTag.first(two);
+        if (countryOne.isPresent() && countryTwo.isPresent())
+        {
+            return countryOne.get().equals(countryTwo.get());
+        }
+
+        throw new CoreException(
+                "All raw Atlas lines must have a country code by the time Relation slicing is done. One of the two Lines {} or {} does not!",
+                one.getIdentifier(), two.getIdentifier());
+    }
+
+    /**
+     * Given a list of relation members, will generate a new list containing all existing, closed or
+     * non-closed (depending on given parameter) members.
+     *
+     * @param relationIdentifier
+     *            The {@link Relation} identifier whose members we're processing
+     * @param members
+     *            The list of {@link RelationMember} we want to group
+     * @param closed
+     *            {@code true} if we want to find closed member lines, false otherwise
+     * @return a subset of the given {@link RelationMember}s, either closed or non-closed
+     */
+    private List<RelationMember> generateMemberList(final long relationIdentifier,
+            final List<RelationMember> members, final boolean closed)
+    {
+        if (members != null && !members.isEmpty())
+        {
+            // Check if all outer ways are closed. If there are any that aren't closed, we need to
+            // build LineRings and fix the gaps
+            return members.stream().filter(member ->
+            {
+                final Line line = this.rawAtlas.line(member.getEntity().getIdentifier());
+                if (line == null)
+                {
+                    logger.error("Line member {} for Relation {} is not the in raw Atlas.",
+                            member.getEntity().getIdentifier(), relationIdentifier);
+                }
+
+                if (!closed)
+                {
+                    return line != null && !line.isClosed();
+                }
+                else
+                {
+                    return line != null && line.isClosed();
+                }
+            }).collect(Collectors.toList());
+        }
+        else
+        {
+            return Collections.emptyList();
+        }
+    }
+
+    /**
+     * Takes in a list of relation members and groups them by country. The output is a map of
+     * country to list of members.
+     *
+     * @param members
+     *            The members we wish to group
+     */
+    private Map<String, List<TemporaryRelationMember>> groupRelationMembersByCountry(
+            final List<TemporaryRelationMember> members)
+    {
+        final Map<String, List<TemporaryRelationMember>> countryEntityMap = members.stream()
+                .collect(Collectors.groupingBy(member ->
+                {
+                    final AtlasEntity entity;
+                    if (member.getType() == ItemType.POINT)
```

Comment:
consider using a switch statement instead?

Reply:
Yeah, that will look a little cleaner!